src/components/flyout/flyout.styles.ts

```diff
@@ -0,0 +1,260 @@
+      z-index: 3;
+    `,
+    outside: css`
+      // match dropshadow
+      ${euiShadowXLarge(euiThemeContext)};
+      ${logicalCSS('right', 'auto')}
+      ${logicalCSS('left', 0)}
+      // Override the hover and focus transitions of buttons
+      animation: none !important;
+      ${euiBreakpoint(euiThemeContext, ['m', 'xl'])} {
+        transform: translateX(calc(-100% - ${euiTheme.size.l})) !important;
+      }
+      ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+        transform: translateX(calc(-100% - ${euiTheme.size.xs})) !important;
+      }
+    `,
+    inside: css``,
+    outsideLeft: css`
+      ${logicalCSS('right', 0)}
+      ${logicalCSS('left', 'auto')}
+      
+      ${euiBreakpoint(euiThemeContext, ['m', 'xl'])} {
+        transform: translateX(calc(100% + ${euiTheme.size.l})) !important;
+      }
+      ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+        transform: translateX(calc(100% + ${euiTheme.size.xs})) !important;
+      }
+    `,
+  };
+};
+
+export const euiFlyoutStyles = (euiThemeContext: UseEuiTheme) => {
+  const euiTheme = euiThemeContext.euiTheme;
+
+  return {
+    euiFlyout: css`
+      ${euiShadowXLarge(euiThemeContext)};
+      position: fixed;
+      ${logicalCSS('top', 0)}
+      ${logicalCSS('bottom', 0)}
+      ${logicalCSS('height', '100%')}
+      z-index: ${euiTheme.levels.flyout};
+      background: ${euiTheme.colors.emptyShade};
+      display: flex;
+      flex-direction: column;
+      align-items: stretch;
+      &:focus {
+        outline: none;
+      }
+      ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+        ${logicalCSS('max-width', '90vw')}
+      }
+    `,
+
+    // Flyout Sizes
+    s: css`
+      ${composeFlyoutSizing(euiThemeContext, 's')}
+    `,
+    m: css`
+      ${composeFlyoutSizing(euiThemeContext, 'm')}
+    `,
+    l: css`
+      ${composeFlyoutSizing(euiThemeContext, 'l')}
+    `,
+
+    // Side
+    right: css`
+      clip-path: polygon(-50% 0, 100% 0, 100% 100%, -50% 100%);
+      ${logicalCSS('border-left', euiTheme.border.thin)}
+      ${logicalCSS('right', 0)}
+      ${euiCanAnimate} {
+        animation: ${euiFlyout} ${euiTheme.animation.normal}
+          ${euiTheme.animation.resistance};
+      }
+    `,
+    left: css`
+      ${logicalCSS('border-right', euiTheme.border.thin)}
+      ${logicalCSS('left', 0)}
+      clip-path: polygon(0 0, 150% 0, 150% 100%, 0 100%);
+      ${euiCanAnimate} {
+        animation: ${euiFlyoutLeft};
+      }
+    `,
+
+    // Type
+    overlay: css``,
+    push: css`
+      box-shadow: none;
+      clip-path: none;
+      animation-duration: 0s !important; // Don't animate on loading a docked nav
+      ${logicalCSS('border-left', euiTheme.border.thick)}
+      // Make sure the header shadows are above
+      z-index: ${Number(euiTheme.levels.flyout) - 1};
+    `,
+    pushLeft: css`
+      ${logicalCSS('border-left', 'none')}
+      ${logicalCSS('border-right', euiTheme.border.thick)}
+    `,
```

Comment:
Haha you're probably very tired of me asking for all these left/right objects but I just think it's so much cleaner to read 😅 
```suggestion
    push: css`
      box-shadow: none;
      clip-path: none;
      animation-duration: 0s !important; // Don't animate on loading a docked nav
      // Make sure the header shadows are above
      z-index: ${Number(euiTheme.levels.flyout) - 1};
    `,
    pushSide: {
      right: css`
        ${logicalCSS('border-left', euiTheme.border.thick)}
      `,
      left: css`
        ${logicalCSS('border-right', euiTheme.border.thick)}
      `,
    },
```